### ISSUE TITLE
Content-dependent misbehaviour of Iterator.span followed by toSeq

### ISSUE THREAD

**Issue by scabug** [scala/bug#5838]
For certain inputs, the result of Iterator.span, followed by toSeq and toList, is wrong. If span is directly followed by toList, everything is fine.

```scala
def spanA(list: List[Int], x: Int) =
  list.iterator.span(_ == x) match {
    case (a,b) => (a.toList, b.toList)
  }

def spanB(list: List[Int], x: Int) =
  list.iterator.span(_ == x) match {
    case (a,b) => (a.toSeq, b.toSeq) match {
      case (a,b) => (a.toList, b.toList)
    }
  }

println("A1: " + spanA(List(1, 2, 3), 1) )
println("B1: " + spanB(List(1, 2, 3), 1) )

println()

println("A2: " + spanA(List(1, 2, 1), 1) )
println("B2: " + spanB(List(1, 2, 1), 1) )
```

results in

```scala
A1: (List(1),List(2, 3))
B1: (List(1),List(2, 3))

A2: (List(1),List(2, 1))
B2: (List(1, 1),List(2))
```

Interestingly, the misbehaviour only seems to occur here when the last element of the input list equals the first, even though the span operation should never even look at the last element in these example cases.

**Comment by SethTisue** [comment]
still broken in 2.11.0, working in 2.11.8, not sure which exact point release in between the fix landed in

**Comment by Philippus** [comment]
I tried it out, it was fixed in 2.11.7. I think in https://github.com/scala/scala/pull/4530 which would mean this is already covered by a test case. But anyway, since I already wrote the code for these specific examples: https://github.com/scala/scala/pull/10922. Feel free to take it or not. :)

**Comment by som-snytt** [comment]
Fair to say duplicated by https://github.com/scala/bug/issues/9332

Whew, I'm glad that guy wrote a test.
